dhis2/dhis2-core
# feat: add support for hotfix identifiers on update notifications

### Summary:
Add support for the new DHIS2 release JSON format served from this URL: https://releases.dhis2.org/v1/versions/stable.json
Prior to this PR there was only support for detecting patch version, now we also have hot fix versions. This PR adds the ability to parse and detect new hot fixes.

### Manual testing:
* Start the server on a clean database, 
* GET request this: http://localhost:8080/api/systemUpdates?forceVersion=2.37.7
* Login to the server as admin and you should see a new message notification in the message icon on the top bar
* Click the message icon and then click the system notification category, then click the latest unread message
* Observe that it has detected a new hot fix version 2.37.7.1

### Automated tests:
_SystemUpdateNotificationServiceTest_, reads a JSON file representing the version document as per latest version when this PR was made, which is (2.38.1)

### Jira:
[DHIS2-13376](https://jira.dhis2.org/browse/DHIS2-13376) Support for additional hotfix version identifier
